plugins/modules/ec2_dedicated_host.py

```diff
@@ -0,0 +1,479 @@
+    type: str
+  host_recovery:
+    description:
+      - Indicates whether to enable or disable host recovery for the Dedicated Host.
+      - Host recovery is disabled by default.
+    type: str
+    default: 'off'
+    choices: ['on', 'off']
+  outpost_arn:
+    description:
+      - The Amazon Resource Name (ARN) of the Amazon Web Services Outpost on which to allocate the Dedicated Host.
+    type: str
+  host_maintenance:
+    description:
+      - Indicates whether to enable or disable host maintenance for the Dedicated Host.
+    type: str
+    default: 'off'
+    choices: ['on', 'off']
+  asset_ids:
+    description:
+      - The IDs of the Outpost hardware assets on which to allocate the Dedicated Hosts.
+      - Targeting specific hardware assets on an Outpost can help to minimize latency between your workloads.
+      - This parameter is supported only if you specify O(outpost_arn).
+      - If you specify this parameter, you can omit O(quantity).
+      - If you specify both O(asset_ids) and O(quantity), then the value for O(quantity) must be equal to the number of O(asset_ids) specified.
+    type: list
+    elements: str
+  auto_placement:
+    description:
+      - Indicates whether the host accepts any untargeted instance launches that match its instance type configuration,
+        or if it only accepts Host tenancy instance launches that specify its unique host ID.
+    type: str
+    default: 'off'
+    choices: ['on', 'off']
+  client_token:
+    description:
+      - case-sensitive identifier that you provide to ensure the idempotency of the request.
+    type: str
+  quantity:
+    description:
+      - The number of Dedicated Hosts to allocate to your account with these parameters.
+      - Required when O(state=present).
+      - If you are allocating the Dedicated Hosts on an Outpost, and you specify O(asset_ids), you can omit this parameter.
+      - If you specify both O(asset_ids) and O(quantity), then the value that you specify for O(quantity) must be equal to the number of O(asset_ids) specified.
+    type: int
+  availability_zone:
+    description:
+      - The Availability Zone in which to allocate the Dedicated Host.
+      - Required if O(state=present).
+    type: str
+extends_documentation_fragment:
+  - amazon.aws.common.modules
+  - amazon.aws.region.modules
+  - amazon.aws.boto3
+  - amazon.aws.tags
+"""
+
+EXAMPLES = r"""
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Release dedicated hosts
+- name: Release dedicated host
+  amazon.aws.ec2_dedicated_host:
+    host_id: "h-0123456789abcdef0"
+
+# Allocate dedicated host using instance type
+- name: Allocate dedicated host using instance_type
+  amazon.aws.ec2_dedicated_host:
+    availability_zone: us-east-1a
+    quantity: 1
+    instance_type: mac2.metal
+    tags:
+      Scope: Mac
+"""
+
+RETURN = r"""
+host:
+  description: Information about the dedicated host created/updated.
```

Comment:
Can you please include samples for each of the the returned information?

Reply:
updated